universal-application-tool-0.0.1/app/services/applicant/Update.java

```diff
@@ -9,8 +9,12 @@
    * A JSON-style path pointing to a scalar value to update in the applicant's {@link
    * ApplicantData}.
    */
-  public abstract String path();
+  public abstract Path path();
 
   /** The value to update the the applicant's {@link ApplicantData} to. */
-  public abstract String value();
+  public abstract Object value();
```

Comment:
Suggestion (blocking): using `Object` has a pretty big drawback in that you lose type checking altogether. Since the values coming in from the form body encoded in the POST, I recommend sticking with the more specific `String` type here and parsing (not casting) to other types as needed in the `switch` in `stageUpdates(
      ApplicantData applicantData, BlockDefinition blockDefinition, ImmutableSet<Update> updates)`

Reply:
done